### PR TITLE
[RSDK-7322] Move analog writers from Board interface to Analog interface

### PR DESCRIPTION
**Pull request by penguinland**
Everything compiles and the linter is happy, but that's as far as I've tested; I don't have an analog writer to try this out with. Olivia, maybe I could borrow one from you?

### CODE REVIEW

File: components/board/client.go [hunk 1]
```diff
@@ -169,7 +153,22 @@ func (ac *analogClient) Read(ctx context.Context, extra map[string]interface{})
 }
 
 func (ac *analogClient) Write(ctx context.Context, value int, extra map[string]interface{}) error {
-	return errors.New("unimplemented")
+	ext, err := protoutils.StructToStructPb(extra)
+	if err != nil {
+		return err
+	}
+	// the api method is named ReadAnalogReader, it is named differenlty than
```
Comment (oliviamiller) [inline components/board/client.go:160]: comment doesnt seem relevant?

**Review by oliviamiller — APPROVED** [review]
lgtm, I don't have an analog writer but you could test with a fake board.

File: components/board/client.go (hunk 1 above)
Comment (penguinland) [reply components/board/client.go:160]: Whoops! Serves me right for copying and pasting. :confounded: Thanks!

**Comment by penguinland** [comment]
I was unable to get the fake board to work. On closer inspection, it's because `analogTestPin` was the empty string, if you use any other pin it throws away any written value when you read it, and if you try to use that pin it complains that the pin can't be the empty string. So, I made the test pin "1", and then things work great (for this one pin; others continue throwing away the written values any time you try reading it). I still have doubts that having different behavior on `analogTestPin` is a good idea or that throwing away written values on any pin is a good idea, but those are questions for another day.

**Pull request merged**